src/data_hub/dataframe.py

```diff
@@ -31,6 +34,46 @@ class DataFrame(pd.DataFrame):
     _modified_columns: dict = dict()
     """if data is modified in a dataframe, and auto_sync is False, this list will contain the columns that have been modified so that the Deep Origin database can be updated. If an empty list, the Deep Origin database will not be updated, and the dataframe matches the Deep Origin database at the time of creation."""
 
+    _allow_adding_rows: bool = True
+    """If `True`, new rows can be added to the dataframe. If `False`, new rows cannot be added to the dataframe."""
+
+    @property
+    def loc(self):
+        class _LocIndexer:
+            def __init__(self, df):
+                self.df = df
+
+            def __getitem__(self, key):
```

Comment:
this is a LocIndexer, which overrides the superclass class attribute. allows us to define what happens when people slice and index into their dataframe